config.json

```diff
@@ -342,6 +354,24 @@
       "values": [],
       "defaultValue": ""
     },
+    "OptionalToolSource": {
```

Comment:
Sounds good. 

As an assign I really think the amount of argument parsing that is happening in these build scripts is way out of control which is at least part of the trouble with the extra parameter fragility.